src/Kiota.Builder/Export/PublicAPIExportService.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Kiota.Builder.CodeDOM;
+using Kiota.Builder.Configuration;
+using Kiota.Builder.Writers;
+using Kiota.Builder.Writers.CSharp;
+using Kiota.Builder.Writers.Go;
+using Kiota.Builder.Writers.Java;
+using Kiota.Builder.Writers.Php;
+using Kiota.Builder.Writers.Python;
+using Kiota.Builder.Writers.Ruby;
+using Kiota.Builder.Writers.Swift;
+using Kiota.Builder.Writers.TypeScript;
+
+namespace Kiota.Builder.Export;
+
+internal class PublicApiExportService
+{
+    internal PublicApiExportService(GenerationConfiguration generationConfiguration)
+    {
+        ArgumentNullException.ThrowIfNull(generationConfiguration);
+        _outputDirectoryPath = generationConfiguration.OutputPath;
+        _conventionService = GetLanguageConventionServiceFromConfiguration(generationConfiguration);
+    }
+    private readonly string _outputDirectoryPath;
+    private readonly ILanguageConventionService _conventionService;
+    private const string DomExportFileName = "kiota-dom-export.txt";
+    private const string InheritsSymbol = "-->";
+    private const string ImplementsSymbol = "~~>";
+    private const string OptionalSymbol = "?";
+    private const string ParentElementAndChildSeparator = "::";
+
+    internal async Task SerializeDomAsync(CodeNamespace rootNamespace, CancellationToken cancellationToken = default)
+    {
+        var filePath = Path.Combine(_outputDirectoryPath, DomExportFileName);
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task
+        await using var fileStream = File.Create(filePath);
+        await using var streamWriter = new StreamWriter(fileStream);
+#pragma warning restore CA2007 // Consider calling ConfigureAwait on the awaited task
+        var entries = GetEntriesFromDom(rootNamespace).Order(StringComparer.OrdinalIgnoreCase).ToArray();
+        foreach (var entry in entries)
+        {
+            await streamWriter.WriteLineAsync(entry.AsMemory(), cancellationToken).ConfigureAwait(false);
+        }
```

Comment:
```suggestion
        var streamWriter = new StreamWriter(fileStream);
        await using (streamWriter.ConfigureAwait(false))
        {
            var entries = GetEntriesFromDom(rootNamespace).Order(StringComparer.OrdinalIgnoreCase).ToArray();
            foreach (var entry in entries)
            {
                await streamWriter.WriteLineAsync(entry.AsMemory(), cancellationToken).ConfigureAwait(false);
            }
        }
```

Reply:
Out of curiosity, what's the change here? Is this to simple avoid the `#pragma warning restore CA2007 // Consider calling ConfigureAwait on the awaited task` ?

Reply:
Yes. It removes the warning and adds the `ConfigureAwait(false)`